Incorrect icon/token types

I forgot to also add `dist/esm/index.d.ts` files in #6192. Without them you'll see these errors:

```sh
➜  npmcrap git:(master) ✗ yarn tsc --jsx react --esModuleInterop --skipLibCheck test.tsx
yarn run v1.22.11
$ /home/thesm/src/npmcrap/node_modules/.bin/tsc --jsx react --esModuleInterop --skipLibCheck test.tsx
test.tsx:2:10 - error TS2305: Module '"@patternfly/react-icons"' has no exported member 'InProgressIcon'.

2 import { InProgressIcon, MaintenanceIcon } from '@patternfly/react-icons';
```

--- a/packages/react-icons/scripts/writeIcons.js
+++ b/packages/react-icons/scripts/writeIcons.js
@@ -92,6 +92,7 @@ function writeIcons(icons) {
     .sort()
     .join('\n');
   outputFileSync(join(outDir, 'esm', 'icons/index.js'), esmIndexString);
+  outputFileSync(join(outDir, 'esm', 'icons/index.d.ts'), esmIndexString);
   outputFileSync(join(outDir, 'js', 'icons/index.d.ts'), esmIndexString);
   outputFileSync(
     join(outDir, 'js', 'icons/index.js'),

--- a/packages/react-tokens/scripts/writeTokens.js
+++ b/packages/react-tokens/scripts/writeTokens.js
@@ -40,7 +40,6 @@ const componentIndex = [];
 const outputIndex = (index, indexFile) => {
   const esmIndexString = index.map(file => `export { ${file} } from './${file}';`).join('\n');
   outputFileSync(join(outDir, 'esm', indexFile), esmIndexString);
-  outputFileSync(join(outDir, 'js', indexFile.replace('.js', '.d.ts')), esmIndexString);
   outputFileSync(
     join(outDir, 'js', indexFile),
     `
@@ -52,6 +51,8 @@ exports.__esModule = true;
 ${index.map(file => `__export(require('./${file}'));`).join('\n')}
 `.trim()
   );
+  outputFileSync(join(outDir, 'esm', indexFile.replace('.js', '.d.ts')), esmIndexString);
+  outputFileSync(join(outDir, 'js', indexFile.replace('.js', '.d.ts')), esmIndexString);
 };
 
 /**
@@ -92,7 +93,7 @@ function writeTokens(tokens) {
   outputIndex(componentIndex, 'componentIndex.js');
 
   // eslint-disable-next-line no-console
-  console.log('Wrote', Object.keys(allIndex).length * 3 + 3, 'token files');
+  console.log('Wrote', Object.keys(allIndex).length * 3 + 4, 'token files');
 }
 
 writeTokens(generateTokens());

--- a/packages/react-tokens/scripts/writeTokens.js
+++ b/packages/react-tokens/scripts/writeTokens.js
@@ -93,7 +93,7 @@ function writeTokens(tokens) {
   outputIndex(componentIndex, 'componentIndex.js');
 
   // eslint-disable-next-line no-console
-  console.log('Wrote', Object.keys(allIndex).length * 3 + 4, 'token files');
+  console.log('Wrote', Object.keys(allIndex).length * 4 + 4, 'token files');
 }
 
 writeTokens(generateTokens());

